python-checks/src/main/java/org/sonar/python/checks/FlaskHardCodedSecretCheck.java

```diff
@@ -0,0 +1,183 @@
+      .map(Symbol::fullyQualifiedName)
+      .filter(FLASK_APP_CONFIG_QUALIFIER_FQNS::contains)
+      .ifPresent(fqn -> verifyUpdateCallArgument(ctx, callExpression));
+  }
+
+  private static void verifyUpdateCallArgument(SubscriptionContext ctx, CallExpression callExpression) {
+    Optional.of(callExpression.arguments())
+      .filter(arguments -> arguments.size() == 1)
+      .map(arguments -> arguments.get(0))
+      .flatMap(TreeUtils.toOptionalInstanceOfMapper(RegularArgument.class))
+      .map(RegularArgument::expression)
+      .map(FlaskHardCodedSecretCheck::getAssignedValue)
+      .filter(FlaskHardCodedSecretCheck::isIllegalDictArgument)
+      .ifPresent(expr -> ctx.addIssue(callExpression, MESSAGE));
+
+  }
+
+  private static Expression getAssignedValue(Expression expression) {
+    if (expression.is(Tree.Kind.NAME)) {
+      return Expressions.singleAssignedValue((Name) expression);
+    }
+    return expression;
+  }
+
+  private static boolean isIllegalDictArgument(Expression expression) {
+    if (expression.is(Tree.Kind.CALL_EXPR)) {
+      return isCallToDictConstructor((CallExpression) expression) && hasIllegalKeywordArgument((CallExpression) expression);
+    } else if (expression.is(Tree.Kind.DICTIONARY_LITERAL)) {
+      return hasIllegalKeyValuePair((DictionaryLiteral) expression);
+    }
+    return false;
+  }
+
+  private static boolean isCallToDictConstructor(CallExpression callExpression) {
+    return Optional.of(callExpression)
+      .map(CallExpression::callee)
+      .flatMap(TreeUtils.toOptionalInstanceOfMapper(Name.class))
+      .map(Name::symbol)
+      .map(Symbol::fullyQualifiedName)
+      .filter("dict"::equals)
+      .isPresent();
+  }
+
+  private static boolean hasIllegalKeyValuePair(DictionaryLiteral dictionaryLiteral) {
+    return dictionaryLiteral.elements().stream()
+      .filter(KeyValuePair.class::isInstance)
+      .map(KeyValuePair.class::cast)
+      .map(KeyValuePair::key)
+      .filter(StringLiteral.class::isInstance)
+      .map(StringLiteral.class::cast)
+      .map(StringLiteral::trimmedQuotesValue)
+      .anyMatch(SECRET_KEY_KEYWORD::equals);
+  }
+
+  private static boolean hasIllegalKeywordArgument(CallExpression callExpression) {
+    return Optional.ofNullable(TreeUtils.argumentByKeyword(SECRET_KEY_KEYWORD, callExpression.arguments()))
+      .map(RegularArgument::expression)
+      .filter(FlaskHardCodedSecretCheck::isStringLiteral)
+      .isPresent();
+  }
+
+  private static void verifyAssignmentStatement(SubscriptionContext ctx) {
+    AssignmentStatement assignmentStatementTree = (AssignmentStatement) ctx.syntaxNode();
+    for (ExpressionList lhsExpression : assignmentStatementTree.lhsExpressions()) {
+      boolean isModifyingSensitiveProperty = lhsExpression.expressions().stream()
+      .anyMatch(FlaskHardCodedSecretCheck::isSensitiveProperty);
+      if (isModifyingSensitiveProperty) {
+        ctx.addIssue(lhsExpression, MESSAGE);
+      }
```

Comment:
Could that be replaced with a `filter` and a `ifPresent`?

Reply:
Since I am working on a stream, I can't invoke `ifPresent`. I will still look for a neater option.

Reply:
I changed the implementation slightly.